src/taipy/core/scenario/scenario.py

```diff
@@ -337,6 +349,24 @@ def remove_tag(self, tag: str):
 
         return tp.untag(self, tag)
 
+    def _get_inputs(self) -> Set[DataNode]:
+        dag = self.__build_dag()
+        return {node for node, degree in dict(dag.in_degree).items() if degree == 0 and isinstance(node, DataNode)}
+
+    def __build_dag(self):
```

Comment:
I think we can reuse the build_dag function from pipeline.py and by making it a static function or a common function in `common.py` What do you think?

Reply:
You are right. Thx.